private-templates-service/client/src/components/Common/VersionModal/VersionModal.tsx

```diff
@@ -115,27 +115,32 @@ class VersionModal extends React.Component<Props, State> {
                 <CardHeaderText>Version</CardHeaderText>
                 <CardHeaderText>Published</CardHeaderText>
                 <CardHeaderText>Status</CardHeaderText>
-                <CardHeaderText>{`${this.state.versionList.filter(function(s) { return s; }).length} Selected`}</CardHeaderText>
+                <CardHeaderText>{`${this.state.versionList.filter(function (s) { return s; }).length} Selected`}</CardHeaderText>
               </CardHeaderRow>
               <CardBody>
-              {this.props.template.instances && this.props.template.instances.map((instance: TemplateInstance, index: number) => (
-            <VersionCardRow>
-              <VersionWrapper>
-                {instance.version}
-              </VersionWrapper>      
-              <DateWrapper>{instance.publishedAt? getDateString(instance.publishedAt) : "Not published"}</DateWrapper>
-              <StatusWrapper>
-                <StatusIndicator state={instance.state}/>
-                <Status>{instance.state && instance.state.toString().charAt(0).toUpperCase() + instance.state.toString().slice(1)}</Status>
-              </StatusWrapper>
-              <CheckboxWrapper><Checkbox checked={this.state.versionList[index]} 
-                onChange={() => {
-                  let updatedVersion = this.state.versionList;
-                  updatedVersion[index] = !updatedVersion[index];
-                  this.setState({ versionList: updatedVersion });
-              }}/></CheckboxWrapper>
-            </VersionCardRow>
-            ))}     
+                {this.props.template.instances && this.props.template.instances.map((instance: TemplateInstance, index: number) => (
+                  <VersionCardRow>
+                    <VersionWrapper>
+                      {instance.version}
+                    </VersionWrapper>
+                    <DateWrapper>{instance.publishedAt ? getDateString(instance.publishedAt) : "Not published"}</DateWrapper>
+                    <StatusWrapper>
+                      <StatusIndicator state={
+                        instance.state
+                          ? instance.state
+                          // should never reach else statement
+                          : PostedTemplate.StateEnum.Draft
```

Comment:
If you know for sure that `instance.state` exists, then enter `instance.state!`

Reply:
It lets the type checker you are confident that instance.state is not null

Reply:
that's cool, didn't know that

Reply:
cool, thx!